llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -2596,6 +2596,15 @@ template <typename T> using has_sizeof = decltype(sizeof(T));
 template <typename T>
 constexpr bool is_incomplete_v = !is_detected<detail::has_sizeof, T>::value;
 
+//===----------------------------------------------------------------------===//
+//     Extra additions to <variant>
+//===----------------------------------------------------------------------===//
+
+template <class... Ts> struct overloaded : Ts... {
+  using Ts::operator()...;
+};
+template <class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
+
```

Comment:
Can you put this in a separate PR? This way we won't have to rebuild the whole tree if this larger PR gets reverted.

Also, doesn't this duplicate `llvm::make_visitor`? Maybe we could merge the two.

Reply:
Yeah, I can split this out.

I don't see `make_visitor` in the codebase (git grep). I looked at all uses of `std::visit` to see if we already had something similar and didn't see anything.

Reply:
https://github.com/llvm/llvm-project/blob/f0f8434afac2d30ac143250377fb6433c68fc0a8/llvm/include/llvm/ADT/STLExtras.h#L1518-L1549

Reply:
Ah, thanks! I can use that instead. Surprised there are no uses of it (at least in LLVM or Clang).

Reply:
I wonder if we should switch to your implementation instead -- seems much simpler

Reply:
I just played around with it in [compiler explorer](https://godbolt.org/z/c7baxEjWP) and I'm not sure why it was implemented that way. Aggregate initialization just does the right thing. It does require using `{}` instead of `()` (until C++20), but I think that's the only difference. I'll just remove it from this patch and if we want to change makeVisitor that can happen separately.